miss_islington/status_change.py

```diff
@@ -100,16 +100,22 @@ async def check_ci_status_and_approval(
 
             title_match = TITLE_RE.match(normalized_pr_title)
             if title_match or is_automerge:
-                success = result["state"] == "success" and not any(
+                failure = any(
                     elem in [None, "failure", "timed_out"]
                     for elem in all_check_run_conclusions
                 )
+                success = result["state"] == "success" and not failure
                 if leave_comment:
                     if success:
                         emoji = "✅"
+                        status = "it's a success"
+                    if failure:
+                        emoji = "❌"
+                        status = "it's a failure or timed out"
                     else:
                         emoji = "❌"
-                    message = f"Status check is done, and it's a {result['state']} {emoji} ."
+                        status = "it's a failure"
```

Comment:
Now that I think about this, this seem to be faulty `if ... if` logic.

If `success` is `True` `failure` is None and thus we will always go into the third path.

I'll open a PR to address this tomorrow. I'll tag you for a review so we can fix this and remove confusion from the `cpython` repo.